test/core/security/credentials_test.cc

```diff
@@ -506,14 +508,13 @@ static void check_request_metadata(void* arg, grpc_error_handle error) {
   check_metadata(state->expected, &state->md_array);
   grpc_credentials_mdelem_array_destroy(&state->md_array);
   grpc_pollset_set_destroy(grpc_polling_entity_pollset_set(&state->pollent));
-  gpr_free(state);
+  delete state;
 }
 
 static request_metadata_state* make_request_metadata_state(
     grpc_error_handle expected_error, const expected_md* expected,
     size_t expected_size) {
-  request_metadata_state* state =
```

Comment:
Hmm, replacing gpr_zalloc directly with new does not seem very safe. Please make sure that all fields are properly initialized (including the C-style structs)